### DISCUSSION TITLE
camera_num = self-global_camera_info() [camera_num] ['Num'] index out of bounds error

### DISCUSSION THREAD

**Discussion by EmisPants** [raspberrypi/picamera2#1091] [Q&A]
when i try to open a preview of my camera I get the following

`[0:25:20.161262316] [2415] INFO Camera camera_manager.opp:313 libcamera vo.3.0+65-6ddd79b5 
Traceback (most recent call last): File "/home/emily/Public/myVenv/untitled.py", line 4, in ‹module> picam2 = Picamera2( ) 
File "/usr/lib/python3/dist-packages/picamera2/picamera2.py", line 246, in —init— camera_num = self-global_camera_info() [camera_num] ['Num'] 
IndexError: list index out of range I 
(program exited with code: 1) Press return to continue`

How do I fix this/what does this issue entail? I am using an HQ camera clone from arducam, this one: [https://www.amazon.com/gp/product/B09YHN5DBY/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1]

I've tried the config.txt file, but this camera is supposedly native so I shouldn't have to change anything in there and get errors when I do.

Thanks!

**Comment by zh529** [comment]
怎么解决的呢

**Comment by davidplowman** [comment]
In config.txt you might need to change
```
camera_auto_detect=1
```
to
```
camera_auto_detect=0
```
and add
```
dtoverlay=imx477
```
(and then reboot).

To test whether the camera is being recognised, type `rpicam-hello --list-cameras` in a terminal window.
